.ci/setup_env_centos.sh

```diff
@@ -58,29 +61,47 @@ declare -A packages=( \
 	[redis]="redis" \
 )
 
-pkgs_to_install=${packages[@]}
-
-for j in ${packages[@]}; do
-	pkgs=$(echo "$j")
-	info "The following package will be installed: $pkgs"
-	pkgs_to_install+=" $pkgs"
-done
-chronic sudo -E yum -y install $pkgs_to_install
-
-if [ "$(arch)" == "x86_64" ]; then
-	echo "Install Kata Containers OBS repository"
-	obs_url="${KATA_OBS_REPO_BASE}/CentOS_${VERSION_ID}/home:katacontainers:releases:$(arch):master.repo"
-	sudo -E VERSION_ID=$VERSION_ID yum-config-manager --add-repo "$obs_url"
-	repo_file="/etc/yum.repos.d/home\:katacontainers\:releases\:$(arch)\:master.repo"
-	sudo bash -c "echo timeout=10 >> $repo_file"
-	sudo bash -c "echo retries=2 >> $repo_file"
-fi
+main()
+{
+	local setup_type="$1"
+	[ -z "$setup_type" ] && die "need setup type"
 
-echo "Install GNU parallel"
-# GNU parallel not available in Centos repos, so build it instead.
-build_install_parallel
+	local pkgs_to_install
+	local pkgs
 
-if [ "$KATA_KSM_THROTTLER" == "yes" ]; then
-	echo "Install ${KATA_KSM_THROTTLER_JOB}"
-	chronic sudo -E yum install ${KATA_KSM_THROTTLER_JOB}
-fi
+	for pkgs in "${minimal_packages[@]}"; do
+		info "The following package will be installed: $pkgs"
+		pkgs_to_install+=" $pkgs"
+	done
+
+	if [ "$setup_type" = "default" ]; then
+		for pkgs in "${packages[@]}"; do
+			info "The following package will be installed: $pkgs"
+			pkgs_to_install+=" $pkgs"
+		done
+	fi
+
+	chronic sudo -E yum -y install $pkgs_to_install
+
+	[ "$setup_type" = "minimal" ] && exit 0
+
+	if [ "$(arch)" == "x86_64" ]; then
+		echo "Install Kata Containers OBS repository"
+		obs_url="${KATA_OBS_REPO_BASE}/CentOS_${VERSION_ID}/home:katacontainers:releases:$(arch):master.repo"
```

Comment:
Good question. I think that is out of scope of this PR as I've just wrapped a `main()` around this existing code but I agree we should either be using packagecloud, or as you say a package cache to avoid hitting the network.

Reply:
Issue #1739 raised for this.